src/internal/core.rs

```diff
@@ -61,6 +64,30 @@ impl<P: Package, V: Version> State<P, V> {
             Rc::make_mut(&mut self.incompatibilities),
         );
     }
+
+    /// Add an incompatibility to the state.
+    pub fn add_incompatibility_from_dependencies(
+        &mut self,
+        package: P,
+        version: V,
+        deps: &DependencyConstraints<P, V>,
+    ) -> RangeFrom<IncompId<P, V>> {
+        let out = self.incompatibility_store.next_id()..;
```

Comment:
Isn't this going to be perilous if the list of dependencies is empty? The `next_id()` function doesn't seem very safe to me.

Reply:
It seems to be fine actually returning an empty slice
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=89f3e3c237bc23df3e501804e08bc7f7